dotnet/src/webdriver/WebDriver.cs

```diff
@@ -679,7 +679,7 @@ protected void StartSession(ICapabilities capabilities)
 
             ReturnedCapabilities returnedCapabilities = new ReturnedCapabilities(rawCapabilities);
             this.capabilities = returnedCapabilities;
-            this.sessionId = new SessionId(response.SessionId);
+            this.sessionId = response.SessionId is null ? null : new SessionId(response.SessionId);
```

Comment:
Interesting, I am reading it as "StartSession(...)" might return null as session id. If `this.sessionId` is null, then is it normal?.. I don't think so.

Reply:
That's a much bigger discussion 😁 

Yes, right now `Response.SessionId` can be null. It will need a greater refactor to be fixed, if it's even possible to *always* populate that field.

If we succeed in that mission, then this change can be reverted. But currently, this will throw an exception with the new null check.

Reply:
Related: #14839

Reply:
`response.SessionId` might be null, I can understand.

Reply:
No, I understand nothing. It is a mess. I believe we will win.

Reply:
Bit by bit. After `SessionId` is strengthened, after the `Response` PR modernizes the type, we will keep improving each little part.

Reply:
OK, I am skipping this PR for a while, I think `SessionId` and even `Response` classes are redundant (if we are talking about big refactoring).

Reply:
I think this part is still good! We don't need to revolutionize the whole process, this class is still useful (and used by users). This PR is small and if we really want to delete it in the future, then we can do that.

This change is mostly for the nullable reference type improvements, which is a lot more short-term than any great refactor. When we turn on nullability project-wide, this class will still need to be addressed.

Reply:
PR is updated to include only nullability improvements.

Reply:
Let's discard this change at all, it is misleading.

Reply:
Unfortunately without this change, we will hit Argument Null exceptions sometimes. The session ID string can be null something.

Reply:
Seems this discussion is obsolete. Executing CI.